README.md

```diff
@@ -100,22 +104,32 @@ Missing from the above list is one final, advanced, trait: `HashAndContext` whic
 
 ### In 1.1
 
-This release includes speed improvements.
-
-- `HashVersion{1}` benefits from some limited speed improvements.
-- `HashVersion{2}` is a new hash context that can be faster (~x100) than
-  `HashVersion{1}`; favor it over `HashVersion{1}` in all cases. Since this version changes
-  the hash values of some objects, `HashVersion{1}` is still the default to avoid breaking
-  existing code. 
-- The requirements for `HashVersion{2}` on the passed hash function have been relaxed, such
-that `alg=crc32` should again work (no need to call `alg=(x,s=UInt32(0)) ->
-crc32c(copy(x),s)`).
-- `root_version`: Most users can safely ignore this function. You only need to define
-`root_version` if you are implementing a context that defines 
-`parent_context(x::MyContext) = nothing` (see `parent_context` details on root contexts). It
-indicates what version of the trait implementations to use (1 or 2). It defaults to 1 to
-avoid changing the hash values of exisitng root contexts, but should be defined to return 2
-to make use of the more optimizied implementationsa. 
+This release includes speed improvements of about 100 fold.
+
+- **Feature**: `HashVersion{2}` is a new hash context that can be up to ~100x faster than
+  `HashVersion{1}`.
+- **Feature**: The requirements for `HashVersion{2}` on the passed hash function have been
+  relaxed, such that `alg=crc32` should again work (no need to call `alg=(x,s=UInt32(0)) ->
+  crc32c(copy(x),s)`).
+- **Feature**: `@ConstantHash` allow for precomputed hash values of constant strings and
+  numbers.
+- **Feature**:  `stable_typename_id` and `stable_type_id` provide compile-time 64 bit hashes
+  of the types of objects
+- **Feature**: `root_version`: Most users can safely ignore this new function. If you are
+  implementing a root context (one that returns `parent_context(::MyContext) = nothing`) you
+  will need to define this function. It indicates what version of the hashing
+  implementations to use (1 or 2). It defaults to 1 to avoid changing the hash values of
+  existing root contexts, but should be defined to return 2 to make use of the more
+  optimized implementations used by `HashVersion{2}`.
+- **Deprecation**: `HashVersion{1}` has been deprecated, favor version 2 over 1 in all cases
+  where backwards compatibility is not required.  
+- **Deprecation**: `qualified_name` and `qualified_type` have been deprected, in favor of
+  `stable_typename_id` and `stable_type_id`.
+- **Deprecation**: `ConstantHash` has been deprecated in favor of the more efficient
+  `@ConstantHash`. Note that if the argument to `ConstantHash` was an expression rather than
+  a compile time constant you would need to replace `ConstantHash(exp)` with `FnHash(_ ->
+  exp)` (but this is probably a code smell, since `hash_method` values should normally only
+  depend on the type of their arguments).
```

Comment:
Sound like I need to clarify. It remains strue that you can call `ConstantHash` with an expression. But if you change `ConstantHash` to `@ConstantHash` and the argument is an expression rather than a constant, it will no longer work. So, while removing deprecated API, the user needs to change `ConstantHash(exp)` to `FnHash(_ -> exp)`

Reply:
gotcha! I would just clarify that then with :

> Note that if the argument to `ConstantHash` was an expression rather than
  a compile time constant you would need to replace `ConstantHash(exp)` with `@ContextHash(FnHash(_ ->
  exp))` when removing the deprecated method.